job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -291,13 +299,39 @@ public void setQuietPeriod(Integer seconds) throws IOException {
     }
 
     @Override public CauseOfBlockage getCauseOfBlockage() {
-        // TODO is there any legitimate reason to block a flow? maybe !isConcurrentBuild()?
+        if (isLogUpdated() && !isConcurrentBuild()) {
```

Comment:
> If build10 finishes before build9

I think the idea is that this should be an impossible situation, assuming `isConcurrentBuild()` when build 10 was scheduled—it should have waited for 9 to finish.

At any rate, here I am just using the same logic as in `AbstractProject` so it is probably good enough. In the future it might be better to pull this up into `Job` or refactor into a helper.